flink-compiler/src/main/java/org/apache/flink/compiler/dag/AbstractPartialSolutionNode.java

```diff
@@ -86,8 +87,8 @@ public void computeInterestingPropertiesForInputs(CostEstimator estimator) {
 	}
 
 	@Override
-	public boolean isFieldConstant(int input, int fieldNumber) {
-		return false;
+	public SemanticProperties getSemanticProperties() {
+		return null;
```

Comment:
Can we improve null pointer safety by returning a special constant SemanticProperties object that marks all fields as changed?